### PR TITLE
Adding JWT option, rearranging

### PR DESCRIPTION
**Pull request by martinthomson**

### CODE REVIEW

File: draft-ietf-webpush-vapid.md [hunk 1]
```diff
@@ -158,18 +168,54 @@ This information would be repeated with each request, but that overhead is
...
+## Client-Vended Tokens {#client-token}
+
+In this model, clients generates a token that it uses to prove ownership over a
+private key.  Use of the same key over time establishes a continuous identity.
+
+Push message requests can be accompanied by a JSON Web Token (JWT) [RFC7519].
+This token is not bound to the request.  As a result, the token is reusable over
+a limited time, based on the value of the "exp" (Expiry) claim in the token.  An
+"exp" claim MUST be included; similarly, an "aud" (Audience) claim MUST include
+the string "webpush".
```
Comment (ckarlof) [inline draft-ietf-webpush-vapid.md:180]: Should the Audience instead be the domain of the push endpoint URL to mitigate stolen token re-use across Push providers?

File: draft-ietf-webpush-vapid.md [hunk 2]
```diff
@@ -158,18 +168,54 @@ This information would be repeated with each request, but that overhead is
...
+(as in {{from}}) or the JWT could include claims about identity.  For the
+latter, the "sub" (subject) claim could include a contact URI for the
+application server.
+
+The JWT MUST use a JSON Web Signature (JWS) [RFC7515].  Both the JWS and JWK
+MUST use an elliptic curve digital signature (ECDSA) key on the NIST P-256 curve
+[FIPS186].
+
+
+## Contact Information Header Field {#from}
```
Comment (ckarlof) [inline draft-ietf-webpush-vapid.md:206]: How do you feel about including the contact email inside the JWT?

File: draft-ietf-webpush-vapid.md (hunk 1 above)
Comment (martinthomson) [inline draft-ietf-webpush-vapid.md:180]: That's a good idea.  No point in making these more generically useful.

File: draft-ietf-webpush-vapid.md (hunk 2 above)
Comment (martinthomson) [inline draft-ietf-webpush-vapid.md:206]: That is fine too.  I'll add the option.

**Pull request merged**